src/main/java/kai/KaiParser.java

```diff
@@ -0,0 +1,207 @@
+import kai.commands.UnmarkCommand;
+import kai.tasks.Deadline;
+import kai.tasks.Event;
+import kai.tasks.Task;
+import kai.tasks.ToDo;
+
+/**
+ * Parser handles the logic of parsing input in one place
+ */
+public class KaiParser {
+    private static final DateTimeFormatter INPUT_FORMATTER =
+            DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    private static final DateTimeFormatter STORAGE_FORMATTER =
+            DateTimeFormatter.ofPattern("MMM dd yyyy");
+
+    /**
+     * Parses a string representing a Task and transforms it into an actual Task
+     * (Note: Input validation is NOT proofed against save file editing)
+     *
+     * @param state the string representation of the Task
+     * @return the Task corresponding to that representation
+     * @throws IllegalArgumentException if the stored Task state is invalid
+     * @throws StringIndexOutOfBoundsException if the stored Task state is invalid
+     * @throws DateTimeParseException if the stored Task state is invalid
+     */
+    public Task parseStoredTask(String state)
+            throws IllegalArgumentException, StringIndexOutOfBoundsException, DateTimeParseException {
+        String type = state.substring(0, state.indexOf(" | ") + 3);
+        state = state.substring(state.indexOf(" | ") + 3);
+
+        boolean isDone = (state.substring(0, state.indexOf(" | ")).equals("1"));
+        state = state.substring(state.indexOf(" | ") + 3);
+
+        String desc;
+        Task res;
+        switch (type) {
+        case "T | ":
+            desc = state;
+            res = new ToDo(desc);
+            break;
+        case "D | ":
+            desc = state.substring(0, state.indexOf(" | "));
+            state = state.substring(state.indexOf(" | ") + 3);
+
+            LocalDate deadline = LocalDate.parse(state, STORAGE_FORMATTER);
+            res = new Deadline(desc, deadline);
+            break;
+        case "E | ":
+            desc = state.substring(0, state.indexOf(" | "));
+            state = state.substring(state.indexOf(" | ") + 3);
+
+            LocalDate from = LocalDate.parse(
+                    state.substring(0, state.indexOf(" | ")), STORAGE_FORMATTER);
+            state = state.substring(state.indexOf(" | ") + 3);
+
+            LocalDate to = LocalDate.parse(state, STORAGE_FORMATTER);
+            res = new Event(desc, from, to);
+            break;
+        default:
+            throw new IllegalArgumentException();
+        }
+
+        if (isDone) {
+            res.setComplete();
+        } else {
+            res.setIncomplete();
+        }
+
+        return res;
+    }
+
+    /**
+     * Parses the command given and takes the appropriate response thereof
+     *
+     * @param input the string corresponding to the command in question
+     * @param taskList the TaskList some of these commands work on
+     * @return the Command to be invoked later on
+     */
+    public Command parseCommand(String input, TaskList taskList) {
```

Comment:
Should some branches be extracted out to reduce verbosity of code?

Reply:
Wdym? parseExitCommand etc, or more likely parseCreateTaskCommand factored out?
Didn't do it because I think the amount of code will remain the same with more JavaDoc boilerplate, or did you have something else in mind?